src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```
@@ -0,0 +1,30 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
```

Constructor does not validate the incoming dependency. Per the guidance to handle null parameters (e.g., "Null parameters"), add a null-check: `if (operationStrategy == null) { throw new RuntimeException("OperationStrategy cannot be null"); }`. This ensures the service fails fast with an informative message rather than producing an NPE later .